### PR TITLE
document how to gather the audit logs 2

### PR DESCRIPTION
**Pull request by mburke5678**
Replacing https://github.com/openshift/openshift-docs/pull/14381

https://jira.coreos.com/browse/MSTR-328
https://github.com/openshift/api/pull/279/files
https://bugzilla.redhat.com/show_bug.cgi?id=1695828

### CODE REVIEW

**Comment by mburke5678** [comment]
@xingxingxia Please take a look.

**Comment by xingxingxia** [comment]
Ying Zhou will help review.

File: modules/nodes-nodes-audit-log-advanced.adoc [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+The advanced audit feature provides several improvements over the
+basic audit functionality, including fine-grained events filtering and multiple output back ends.
+
+To enable the advanced audit feature, provide the following values in the
+`openshift_master_audit_config` parameter:
+
+----
+openshift_master_audit_config={"enabled": true, "auditFilePath": "/var/lib/origin/oscp-audit.log", "maximumFileRetentionDays": 14, "maximumFileSizeMegabytes": 500, "maximumRetainedFiles": 5, "policyFile": "/etc/origin/master/adv-audit.yaml", "logFormat":"json"}
+----
+
```
Comment (zhouying7780) [inline modules/nodes-nodes-audit-log-advanced.adoc]: Where to set the `openshift_master_audit_config` parameter ? How to set it ?

File: modules/nodes-nodes-audit-log-advanced.adoc [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+[IMPORTANT]
+====
+The policy file *_/etc/origin/master/adv-audit.yaml_* must be available on each master node.
```
Comment (zhouying7780) [inline modules/nodes-nodes-audit-log-advanced.adoc]: Now the default `policy File` on each master node is: /openshift.local.audit/policy.yaml, and it's hardcode , could we overwrite the default variable?   And how ?

File: modules/nodes-nodes-audit-log-basic-config.adoc [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+                        "RequestReceived"
+                    ]
+                }
+            ]
+        }
+    },
+
+----
+<1> File path where the requests should be logged to. If not set, logs are printed to master logs.
+<2> A boolean to enable or disable audit logs. Default is `false`.
```
Comment (zhouying7780) [inline modules/nodes-nodes-audit-log-basic-config.adoc]: Now the default is `true`

File: modules/nodes-nodes-audit-log-basic-config.adoc [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+
+. Configure the audit parameters
+
+.. Edit the auditing configuration file on the API server pod:
++
+----
+$ oc -n openshift-kube-apiserver exec <kube-api-server-name> cat /etc/kubernetes/static-pod-resources/configmaps/config/config.yaml |json_reformat 
+----
+
+.. Edit the `auditConfig` stanza as needed:
```
Comment (zhouying7780) [inline modules/nodes-nodes-audit-log-basic-config.adoc]: How to edit ?  Seems all by hard code.

File: modules/nodes-nodes-audit-log-advanced.adoc (hunk 1 above)
Comment (mburke5678) [reply modules/nodes-nodes-audit-log-advanced.adoc]: @zhouying7780 I apologize for the confusion. The advanced audit feature that this file describes is not in 4.1.

**Comment by mburke5678** [comment]
@deads2k I believe you worked on this feature for 4.x (?). Can you help with this PR? I think I have the _Configure the audit parameters_ step wrong. The command I provide only displays the values. How can the user configure them? 
Here is a preview:
http://file.rdu.redhat.com/~mburke/nodes-nodes-audit-log-2/nodes/nodes/nodes-nodes-audit-log.html#nodes-nodes-audit-log-basic-config-nodes-nodes-working

File: modules/nodes-nodes-audit-log-advanced.adoc (hunk 2 above)
Comment (mburke5678) [reply modules/nodes-nodes-audit-log-advanced.adoc]: The advanced audit feature that this file describes is not in 4.1.

File: modules/nodes-nodes-audit-log-basic-config.adoc (hunk 4 above)
Comment (mburke5678) [reply modules/nodes-nodes-audit-log-basic-config.adoc]: @deads2k How does the user edit the audit configuration file? Or, can the user not edit? The above command displays the file, but doesn't allow editing.

**Comment by mburke5678** [comment]
@zhouying7780 According to David Eads in Slack: _there isn't a way to configure it, so if that's the doc page, you dont' need it_.  I commented out the section [you were asking about](https://github.com/openshift/openshift-docs/pull/14480#discussion_r278422580) above.

File: modules/nodes-nodes-audit-log-basic-config.adoc [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+  logLevel: "Normal" <1>
+  operatorloglevel: "TraceAll" <2>
+----
+<1> Specify the log level for an overall component.  
+<2> Specify the log level for the API server Operator.
++
+You can set the log level for both settings to one of the following. The setting can be different for each setting:
++
+* *Normal*. Normal is the default.  Normal working log information, including helpful notices for auditing or common operations. Similar to `glog=2`.
+* *Debug*. Debug is for troubleshooting problems. A greater quanitity of notices than *Normal*, but contain less information thatn *Trace*. Common operations might be logged.  Similar to `glog=4`.
```
Comment (zhouying7780) [inline modules/nodes-nodes-audit-log-basic-config.adoc]: Typo : thatn *Trace*. Should be 'than'

File: modules/nodes-nodes-audit-log-basic-config.adoc [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+* *Trace*. Trace is for troubleshooting problems when *Debug* is not verbose enough. Logs every function call as part of a common operation, including tracing execution of a query.  Similar to `glog=6`.
+* *TraceAll*. TraceAll is troubleshoting at the level of API content/decoding. Contains complete body content. In production clusters, this setting causess performance degradation and results in a significant number of logs. Similar to ` glog=8`.
```
Comment (zhouying7780) [inline modules/nodes-nodes-audit-log-basic-config.adoc]: Typo: causess , should be `causes`

File: modules/nodes-nodes-audit-log-basic-viewing.adoc [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+.. View the {product-title} API server log for a specific master node and timestamp or view all the logs for that master:
++
+----
+$ oc adm node-logs <log-name> --path=openshift-apiserver/audit.log
+----
++
+For example:
++
+----
+$ oc adm node-logs ip-10-0-140-97.ec2.internal audit-2019-04-08T13-09-01.227.log --path=openshift-apiserver/audit.log
```
Comment (zhouying7780) [inline modules/nodes-nodes-audit-log-basic-viewing.adoc]: Should be `oc adm node-logs ip-10-0-140-97.ec2.internal audit-2019-04-08T13-09-01.227.log --path=openshift-apiserver/audit-2019-04-08T13-09-01.227.log`

File: modules/nodes-nodes-audit-log-basic-viewing.adoc [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+.. View the Kubernetes API server log for a specific master node and timestamp or view all the logs for that master:
++
+----
+$ oc adm node-logs <log-name> --path=kube-apiserver/audit.log
+----
++
+For example:
++
+----
+$ oc adm node-logs ip-10-0-140-97.ec2.internal audit-2019-04-09T14-07-27.129.log --path=kube-apiserver/audit.log
```
Comment (zhouying7780) [inline modules/nodes-nodes-audit-log-basic-viewing.adoc]: Should be `oc adm node-logs ip-10-0-140-97.ec2.internal audit-2019-04-09T14-07-27.129.log --path=kube-apiserver/audit-2019-04-09T14-07-27.129.log`

File: nodes/nodes/nodes-nodes-audit-log.adoc [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+// modules required to cover the user story. You can also include other
+// assemblies.
+
+
+include::modules/nodes-nodes-audit-log-basic.adoc[leveloffset=+1]
+
+include::modules/nodes-nodes-audit-log-basic-config.adoc[leveloffset=+1]
+
+include::modules/nodes-nodes-audit-log-basic-viewing.adoc[leveloffset=+1]
+
```
Comment (zhouying7780) [inline nodes/nodes/nodes-nodes-audit-log.adoc]: All the three up urls are unaccessible

**Comment by zhouying7780** [comment]
@mburke5678 , some typos, have commented, other LGTM.

File: nodes/nodes/nodes-nodes-audit-log.adoc (hunk 9 above)
Comment (mburke5678) [reply nodes/nodes/nodes-nodes-audit-log.adoc]: These strings add the files to this document during the publishing process and are not accessible at this in this file directly.

**Comment by mburke5678** [comment]
@zhouying7780 I apologize for the typos. I usually perform a spell check pass before posting. Apparently, I neglected to here. Thank you for catching them!!

**Pull request merged**